### PR TITLE
Add Testing for Django 1.11a1 (WiP for compat)

### PR DESCRIPTION
**Pull request by davidszotten**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,13 +1,14 @@
 [tox]
 envlist =
     {py27,py32,py33,py34,py35}-django18
-    {py27,py34,py35}-{django19,django110,django-latest}
+    {py27,py34,py35}-{django19,django110,django111,django-latest}
```
Comment (carltongibson) [inline tox.ini]: `django-latest` only needs testing against `py35` I think. (2.0 will require this)

**Comment by carltongibson** [comment]
> (Fancy making Travis use tox? As per DRF and Django-Filter examples?)

Not compulsory if you don't fancy it.

**Comment by davidszotten** [comment]
yes, sharing seems reasonable. also probably don't want to actually land this until django 1.11 lands

mostly wanted to just get travis running against the pr

any thoughts on the `TODO` or the l10n issue in #683 ?

File: crispy_forms/utils.py [hunk 2]
```diff
@@ -102,16 +90,19 @@ def render_field(
...
-                            set_hidden(field_instance.widget.widgets[index])
-                            field_instance.widget.widgets[index] = field_instance.hidden_widget()
+                            field_instance.widget.widgets[index] = field_instance.hidden_widget(attr)
 
-                        field_instance.widget.widgets[index].attrs.update(attr)
+                        else:
+                            # TODO: can we just injext when instantiating?
+                            # where does this happen (here for hidden, where
+                            # for regular?)
+                            field_instance.widget.widgets[index].attrs.update(attr)
```
Comment (carltongibson) [inline crispy_forms/utils.py]: Can't see a motivation for a change here. Can you explain?

**Comment by carltongibson** [comment]
For #683 my first though is just to branch on `django.VERSION`...

**Comment by davidszotten** [comment]
so if branching, do we also change the behaviour of the templates like https://github.com/django-crispy-forms/django-crispy-forms/blob/dev/crispy_forms/templates/bootstrap4/layout/radioselect.html#L11

**Comment by carltongibson** [comment]
@davidszotten The quick answer would be, Yes, in theory. 

Ideally we'd maintain backwards compatibility. If that's too much effort though, then I'm happy to consider a breaking change. 

Current release is Good Enough™: it's mature and battle tested. We can call out the change and advise people stick to the old version until they're ready for 1.11. 

If it's not too much effort though I'd rather maintain backwards compatibility. 

What's your take on it? (I need to look in depth myself.)

The 1.11a1 build should be failing right? That it's not means we're missing test coverage here.

File: crispy_forms/tests/test_form_helper.py [hunk 3]
```diff
@@ -436,6 +436,7 @@ def test_disable_csrf():
 
 
 def test_render_hidden_fields():
+    from .utils import contains_partial
```
Comment (carltongibson) [inline crispy_forms/tests/test_form_helper.py]: Why is this import here, rather than at the top of the file?

**Comment by carltongibson** [comment]
This looks good. I'm inclined to merge this — setting up testing against 1.11a1 — and come back to #683

File: crispy_forms/utils.py (hunk 2 above)
Comment (davidszotten) [inline crispy_forms/utils.py]: django 1.11 pops `type` from attrs in `__init__`. Just overriding the attrs later ended up with `type=hidden` declared twice

**Comment by davidszotten** [comment]
do you prefer to keep the branch history or would you like me to squash the pr down to a single commit?

**Comment by carltongibson** [comment]
Good effort. Thank you!
